provider/pkg/gen/overlays.go

```diff
@@ -1230,6 +1230,55 @@ var kustomizeDirectoryResource = pschema.ResourceSpec{
 	},
 }
 
+//go:embed examples/overlays/kustomizeDirectoryV2.md
+var kustomizeDirectoryV2MD string
+
+var kustomizeDirectoryV2Resource = pschema.ResourceSpec{
+	IsComponent: true,
+	ObjectTypeSpec: pschema.ObjectTypeSpec{
+		IsOverlay:   false,
+		Description: kustomizeDirectoryV2MD,
+		Properties: map[string]pschema.PropertySpec{
+			"resources": {
+				TypeSpec: pschema.TypeSpec{
+					Type: "string",
+				},
+				Description: "Resources created by the Directory resource.",
+			},
+		},
+		Type: "object",
+	},
+	InputProperties: map[string]pschema.PropertySpec{
+		"directory": {
+			TypeSpec: pschema.TypeSpec{
+				Type: "string",
+			},
+			Description: "The directory containing the kustomization to apply. The value can be a local directory or a folder in a\ngit repository.\nExample: ./helloWorld\nExample: https://github.com/kubernetes-sigs/kustomize/tree/master/examples/helloWorld",
+		},
+		"namespace": {
+			TypeSpec: pschema.TypeSpec{
+				Type: "string",
+			},
+			Description: "The default namespace. Defaults to the provider's namespace.",
```

Comment:
`The default namespace` may not be super clear. Perhaps something like the namespace to apply manifests?